cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,398 @@
+	HostIPC                bool
+	HostNetwork            bool
+	HostPID                bool
+	HostPorts              bool
+	PrivilegeEscalation    bool // this can be true or false
+	PrivilegedContainer    bool
+	RunAsUser              bool
+	ReadOnlyRootFilesystem bool
+	RunAsNonRoot           bool
+	FsGroup                bool
+	SeLinuxContext         bool
+	Capabilities           string
+	HaveDropCapabilities   bool
+	AllVolumeAllowed       bool
+}
+
+var (
+	requiredDropCapabilities = []string{"MKNOD", "SETUID", "SETGID", "KILL"}
+	dropAll                  = []string{"ALL"}
+	category2AddCapabilities = []string{"NET_ADMIN, NET_RAW"}
+	category3AddCapabilities = []string{"NET_ADMIN, NET_RAW, IPC_LOCK"}
+	Category1                = ContainerSCC{false,
+		false,
+		false,
+		false,
+		false,
+		true,
+		false,
+		true,
+		false,
+		true,
+		true,
+		true,
+		category1,
+		true,
+		true}
+
+	Category1NoUID0 = ContainerSCC{false,
+		false,
+		false,
+		false,
+		false,
+		true,
+		false,
+		false,
+		false,
+		false,
+		true,
+		true,
+		category1,
+		true,
+		true}
+
+	Category2 = ContainerSCC{false,
+		false,
+		false,
+		false,
+		false,
+		true,
+		false,
+		true,
+		false,
+		true,
+		true,
+		true,
+		"category2",
+		true,
+		true}
+
+	Category3 = ContainerSCC{false,
+		false,
+		false,
+		false,
+		false,
+		true,
+		false,
+		true,
+		false,
+		true,
+		true,
+		true,
+		"category3",
+		true,
+		true}
+)
+
+func GetContainerSCC(cut *corev1.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HostPorts = false
+	for _, aPort := range cut.Ports {
+		if aPort.HostPort != 0 {
+			containerSCC.HostPorts = true
+			break
+		}
+	}
+	containerSCC = updateCapabilities(cut, containerSCC)
+	if cut.SecurityContext != nil && cut.SecurityContext.AllowPrivilegeEscalation != nil {
+		logrus.Info("PrivilegeEscalation is true")
+		containerSCC.PrivilegeEscalation = true
+	}
+	containerSCC.PrivilegedContainer = false
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil {
+		containerSCC.PrivilegedContainer = *(cut.SecurityContext.Privileged)
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		logrus.Info("RunAsUser is ", cut.SecurityContext.RunAsUser)
+		containerSCC.RunAsUser = true
+	}
+	containerSCC.ReadOnlyRootFilesystem = false
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil {
+		containerSCC.ReadOnlyRootFilesystem = *cut.SecurityContext.ReadOnlyRootFilesystem
+	}
+	containerSCC.RunAsNonRoot = false
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil {
+		logrus.Info("RunAsNonRoot is ", *(cut.SecurityContext.RunAsNonRoot))
+		containerSCC.RunAsNonRoot = *cut.SecurityContext.RunAsNonRoot
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContext = true
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *corev1.Container, containerSCC ContainerSCC) ContainerSCC {
```

Comment:
I still think that this should just adjust the pointer versus make a copy of a struct.

```suggestion
func updateCapabilities(cut *corev1.Container, containerSCC *ContainerSCC) {
```